core/lib/lantern/page-dependency-graph.js

```diff
@@ -593,14 +593,13 @@ class PageDependencyGraph {
   }
 
   /**
-   * @param {LH.TraceEvent[]} mainThreadEvents
    * @param {LH.Trace} trace
-   * @param {LH.Artifacts.TraceEngineResult} traceEngineResult
-   * @param {LH.Artifacts.URL} URL
+   * @return {Map<number, number[]>}
    */
-  static async createGraphFromTrace(mainThreadEvents, trace, traceEngineResult, URL) {
-    // TODO: trace engine should handle this.
+  static _findServiceWorkerThreads(trace) {
+    // TODO: trace engine should provide a map of service workers, like it does for workers.
```

Comment:
it's possible that map already includes (same-process) serviceworkers… do you know it doesn't?
regardless having pid matching for it too seems like a requirement.